src/HUD.h

```diff
@@ -66,20 +65,88 @@ const std::unordered_map<std::string, std::string> fclassPL = {
+    { "farmyard", "podwórze gospodarskie" },
+    { "swimming_pool", "basen" },
+    { "kindergarten", "przedszkole" },
+    { "school", "szkoła" },
+    { "pitch", "boisko" },
+    { "theatre", "teatr" },
+    { "car_wash", "myjnia" },
+    { "attraction", "atrakcja turystyczna" },
+    { "graveyard", "cmentarz" },
+    { "fountain", "fontanna" },
+    { "bakery", "piekarnia" },
+    { "castle", "zamek" },
+    { "dentist", "dentysta" },
+
+    // ---- Added from CSV ----
+    { "airport", "lotnisko" },
+    { "apron", "płyta lotniska" },
+    { "arts_centre", "centrum sztuki" },
+    { "artwork", "dzieło sztuki" },
+    { "bank", "bank" },
+    { "bar", "bar" },
+    { "beach", "plaża" },
+    { "beauty_shop", "salon kosmetyczny" },
+    { "bench", "ławka" },
+    { "beverages", "napoje" },
+    { "bicycle_rental", "wypożyczalnia rowerów" },
+    { "bicycle_shop", "sklep rowerowy" },
+    { "biergarten", "ogródek piwny" },
+    { "bookshop", "księgarnia" },
+    { "buddhist", "obiekt buddyjski" },
+    { "bus_station", "dworzec autobusowy" },
+    { "butcher", "rzeźnik" },
+    { "cafe", "kawiarnia" },
+    { "camp_site", "pole namiotowe" },
+    { "car_repair", "warsztat samochodowy" },
+    { "car_sharing", "carsharing" },
+    { "charging_station", "stacja ładowania" },
+    { "cinema", "kino" },
+    { "college", "uczelnia" },
+    { "community_centre", "centrum społecznościowe" },
+    { "construction", "teren budowy" },
+    { "convenience", "sklep spożywczy" },
+    { "courthouse", "sąd" },
+    { "fast_food", "fast food" },
+    { "fire_station", "straż pożarna" },
+    { "florist", "kwiaciarnia" },
+    { "fuel", "stacja paliw" },
+    { "hospital", "szpital" },
+    { "hotel", "hotel" },
+    { "information", "informacja" },
+    { "library", "biblioteka" },
+    { "marketplace", "targowisko" },
+    { "museum", "muzeum" },
+    { "parking", "parking" },
+    { "pharmacy", "apteka" },
+    { "place_of_worship", "miejsce kultu" },
+    { "police", "policja" },
+    { "post_office", "poczta" },
+    { "pub", "pub" },
+    { "restaurant", "restauracja" },
+    { "supermarket", "supermarket" },
+    { "taxi", "postój taksówek" },
+    { "townhall", "ratusz" },
+    { "university", "uniwersytet" },
+    { "viewpoint", "punkt widokowy" },
+    { "zoo", "zoo" }
 };
+
 // Create a resize handler class
+// Simplified HUDResizeHandler class - add this to HUD.h
 class HUDResizeHandler : public osgGA::GUIEventHandler {
 public:
-    HUDResizeHandler(osg::Camera* hudCamera, osg::Geode* geode,
-                     const std::string& logoFile, float scale)
-        : _hudCamera(hudCamera), _geode(geode), _logoFile(logoFile),
-          _scale(scale), _textMarginLeft(20.0f), _textMarginTop(40.0f),
-          _logoMarginRight(20.0f), _logoMarginTop(20.0f)
+    HUDResizeHandler(osg::Group* root, const std::string& logoFile, float scale)
+        : _root(root), _logoFile(logoFile), _scale(scale), _hudCamera(nullptr)

```

Comment:
```suggestion
    HUDResizeHandler(osg::Camera* hudCamera, osg::Geode* geode,
                     const std::string& logoFile, float scale)
        : _hudCamera(hudCamera), _geode(geode), _logoFile(logoFile),
          _scale(scale)
    {}
```
There is a strong preference to pass the HUD camera as before, as this is how you can remove the initial camera from the scene.

Reply:
Resolved